### PR TITLE
Proxy edits

### PR DESCRIPTION
**Pull request by samasastudio**
-altered how express.static serves static files at listingId endpoints
-helps pulling bundle for proxy script
-altered tests to run on change in pathname
-altered express middleware to handle invalid listingIds
-all API route tests made to pass
-created getData method to help control unit testing
-added comments
-gitignore not ignoring coverage reports (please disregard)
@andyoe0102 @cmonez @krisdo

### CODE REVIEW

File: client/app.jsx [hunk 1]
```diff
@@ -8,8 +8,9 @@ class App extends React.Component {
     super(props);
 
     // this.requestedId = JSON.parse(window.location.href.split('/')[3]); 
-    this.requestedId = JSON.parse(window.location.pathname.slice(1));
+    this.requestedId = JSON.parse(window.location.pathname.slice(1, -1));
```
Comment (cmonez) [inline client/app.jsx:11]: Just refactored use this method. Although I didn't have to JSON.parse. Is that necessary?

File: client/app.jsx [hunk 2]
```diff
@@ -43,7 +40,11 @@ class App extends React.Component {
       .catch(err => {
         console.log("GET for data failed: ", err);
       })
+  }
 
+  componentDidMount() {
+    //makes call to get data for listing
+    this.getData();
```
Comment (cmonez) [inline client/app.jsx:47]: I really like that you abstracted your data call from componentDidMount. Makes your componentDidMount look super clean. Maybe I should do this, hmmmmmm

File: server/index.js [hunk 3]
```diff
@@ -6,23 +6,33 @@ const Promise = require('bluebird');
...
+  if (req.params.listingId > 99 || req.params.listingId < 0) { 
+    res.sendStatus(404);
+  }
+  next()
+})
+//serve static files at listing sub collection
+app.use('/:listingId/', express.static(__dirname + '../../public'));
 app.use(express.json());
 
+//GET API route for listing data (called once componentDidMount & getData is called)
```
Comment (cmonez) [inline server/index.js:21]: I like your notes. Makes it easy to follow your logic.

Comment (samasastudio) [reply server/index.js:21]: thanks man, trying to be good about doing it.  probably helps get rid of monkey brain lol.

File: client/app.jsx (hunk 1 above)
Comment (samasastudio) [reply client/app.jsx:11]: I seem to remember that I was doing this for my tests at them moment, but probably something that can be refactored later.  Only necessary if you need it.

**Pull request merged**